### PR TITLE
Add persistent "link to wpcom" notice when user not linked (and other notice improvements)

### PR DESCRIPTION
**Pull request by dereksmart**
This adds a persistent notice that will be shown to unlinked **users**, similar to the button that is shown in current Jetpack at the top of the page. 

It looks like this: 
unlinked

It's not dismissible. 

**To Test:** 
- Log into a connected Jetpack site as any unlinked user.  
- You should be able to link to WordPress.com
- You should not see the notice when you are linked
- You should not see the notice when the site is not connected

**Other Improvements to notices**
- Added README about different types of notices we're using.
- Now using proper `NoticeAction` for notice CTA's, rather than inline links.  
- Looks much better: 

Before:
before-1
before-2

After: 
after-1
after-2

### CODE REVIEW

**Comment by MichaelArestad** [comment]
What would it look like as a default notice (not info)?

File: _inc/client/components/jetpack-notices/index.jsx [hunk 1]
```diff
@@ -122,25 +122,70 @@ export const DevModeNotice = React.createClass( {
 
 } );
 
+export const UserUnlinked = React.createClass( {
+	displayName: 'UserUnlinked',
+
+	render() {
+		console.log( this.props );
```
Comment (zinigor) [inline _inc/client/components/jetpack-notices/index.jsx]: `console.log` detected :)

**Comment by zinigor** [comment]
Everything seems to work OK, except for one `console.log` and a minor CSS problem:
grab1

**Comment by MichaelArestad** [comment]
How important is it that we make someone connect? (also, this is the site connection, correct?) I know they lose about half or more of the features, but they still have some.

**Comment by MichaelArestad** [comment]
My point being, if it's that important for them to connect, maybe we should just require a connection rather than bugging them to connect.

**Comment by dereksmart** [comment]
@MichaelArestad it's definitely not important enough to block secondary admin access to Jetpack if not connected.

**Comment by eliorivero** [comment]
Looks good, and the notice action is nice! 🐑

**Pull request merged**